extension/src/pipeline/quickPick.ts

```diff
@@ -52,6 +52,69 @@ const formatFieldQuickPickValues = (
   return formattedFields
 }
 
+const verifyFilesHavingSingleField = (files: {
+  [file: string]: string[] | string
+}) => {
+  for (const [file, fields] of Object.entries(files)) {
+    if (Array.isArray(fields)) {
+      void Toast.showError(
+        `${file} cannot have more than one metric selected. See [an example](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#available-configuration-fields) of a plot with multiple x metrics.`
+      )
+      return
+    }
+  }
+
+  return files
+}
+
+const verifyXFields = (
+  xValues: { file: string; key: string }[] | undefined
+) => {
+  if (!xValues || xValues.length === 0) {
+    return
+  }
+
+  const x = formatFieldQuickPickValues(xValues)
+
+  const doFilesHaveOneField = verifyFilesHavingSingleField(x)
+
+  if (!doFilesHaveOneField) {
+    return
+  }
+
+  return { firstXKey: xValues[0].key, x: x as PlotConfigData['x'] }
+}
+
+const verifyYFields = (
+  yValues: { file: string; key: string }[] | undefined,
+  xFieldsLength: number
+) => {
+  if (!yValues || yValues.length === 0) {
+    return
+  }
+
+  const y = formatFieldQuickPickValues(yValues)
+
+  if (xFieldsLength === 1) {
+    return { firstYKey: yValues[0].key, y }
+  }
+
+  if (yValues.length !== xFieldsLength) {
+    void Toast.showError(
+      'The amount of y metrics needs to match the amount of x metrics. See [an example](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#available-configuration-fields) of a plot with multiple x metrics.'
```

Comment:
[C] When there are multiple x metrics selected there must be an equal number of y metrics.

[Q] Also, did you consider using a version of `quickPickLimitedValues` when there are multiple xs? That would avoid the need for this validation. Thinking out loud => maybe we could have a quick pick that validates the number of items selected in the onDidAccept callback?

Reply:
[F] The CLI error is `WARNING: In 'workspace', cannot have different number of x and y data sources. found 2 x and 4 y data sources.`